// Copyright 2021 Alberto Rodelgo.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package grpcinsurer;

// The insurance service definition.
service GRPCInsurer {

  // Returns a policy holder and its policies by userID
  rpc GetContactAndPoliciesById (UserRequest) returns (PolicyHolderResponse) {}

  // Returns a single policyholder and its policies by MobileNumber
  rpc GetContactsAndPoliciesByMobileNumber (MobileNumberRequest) returns (PolicyHolderResponse) {}
}

// The request message containing the user's ID
message UserRequest {
  int64 id = 1;
}

// The request message containing the mobile number
message MobileNumberRequest {
  string mobile_number = 1;
}

// User definition
message User {
  string name = 1;
  string mobile_number = 2;
}

// Policy definition
message Policy {
  string mobile_number = 1;
  int64 premium = 2;
  string type = 3;
}

// Policy holder (user) and its policies response
message PolicyHolderResponse {
  User policy_holder = 1;
  repeated Policy policies = 2;
}